### PR TITLE
Feature : Force save attributes

### PR DESCRIPTION
**Pull request by justinehell**
- Add checkbox to allow saving attribute field(s) in error
- Add function to handle form validation

### CODE REVIEW

File: src/components/drawer/ComponentDetailPanel.vue [hunk 1]
```diff
@@ -178,6 +185,17 @@ async function save() {
   isVisible.value = false;
 }
 
+/**
+ * Handle form validation.
+ */
+function save() {
+  form.value.validate().then(async (success) => {
```
Comment (Zorin95670) [inline src/components/drawer/ComponentDetailPanel.vue:192]: remove `async`

File: src/components/drawer/ComponentDetailPanel.vue [hunk 2]
```diff
@@ -157,7 +164,7 @@ let viewSwitchSubscription;
 /**
  * Update local component data and emit DrawEvent & RenderEvent events.
  */
-async function save() {
+async function submit() {
```
Comment (Zorin95670) [inline src/components/drawer/ComponentDetailPanel.vue:167]: at the end of this function, please set `forceSave` to false

Comment (justinehell) [reply src/components/drawer/ComponentDetailPanel.vue:167]: done

File: src/components/drawer/ComponentDetailPanel.vue [hunk 3]
```diff
@@ -178,6 +185,17 @@ async function save() {
   isVisible.value = false;
 }
 
+/**
+ * Handle form validation.
+ */
+function save() {
+  form.value.validate().then(async (success) => {
+    if (forceSave.value || success) {
+      await submit();
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/components/drawer/ComponentDetailPanel.vue]: done

File: src/components/drawer/ComponentDetailPanel.vue (hunk 1 above)
Comment (justinehell) [reply src/components/drawer/ComponentDetailPanel.vue]: done

**Review by Zorin95670 — APPROVED** [review]

File: tests/unit/components/drawer/ComponentDetailPanel.spec.js [hunk 4]
```diff
@@ -121,12 +125,49 @@ describe('test component: Plugin Component Detail Panel', () => {
...
+      };
+      wrapper.vm.forceSave = true;
+
+      await wrapper.vm.save();
+      await wrapper.vm.$nextTick();
+
+      expect(pluginRenderNext).toHaveBeenCalledTimes(1);
+    });
+
+    it('should not emit event if form validation is not successful', async () => {
```
Comment (k-y0u) [inline tests/unit/components/drawer/ComponentDetailPanel.spec.js:162]: and forceSave is false

Comment (justinehell) [reply tests/unit/components/drawer/ComponentDetailPanel.spec.js]: done

**Review by k-y0u — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**